extensions.html

```diff
@@ -209,33 +338,356 @@ <h2>
             values.
           </p>
         </dd>
-      </dl>
-    </section>
-    <section>
-      <h2>
-        <dfn>GamepadHapticActuatorType</dfn> Enum
-      </h2>
-      <p>
-        The actuator type determines the force applied for a given
-        <var>value</var> in <code>GamepadHapticActuator.pulse()</code>.
-      </p>
-      <pre class="idl">
-        enum GamepadHapticActuatorType {
-          "vibration"
-        };
-      </pre>
-      <dl data-dfn-for="GamepadHapticActuatorType">
         <dt>
-          <dfn>vibration</dfn>
+          <dfn>reset()</dfn>
         </dt>
         <dd>
-          Vibration is a rumbling effect often implemented as an offset weight
-          driven on a rotational axis. The <code>value</code> of a vibration
-          force determines the frequency of the rumble effect and is normalized
-          between <code>0.0</code> and <code>1.0</code>. The neutral value is
-          <code>0.0</code>.
+          <p>
+            The {{GamepadHapticActuator/reset()}} method steps are:
+          </p>
+          <ol>
+            <li>Let |document| be the [=current settings object=]'s [=relevant
+            global object=]'s [=associated `Document`=].
+            </li>
+            <li>If |document| is `null` or |document| is not [=Document/fully
+            active=] or |document|'s [=visibility state=] is `"hidden"`, return
+            [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
+            </li>
+            <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+            is not `undefined`, [=queue a global task=] on the [=relevant
+            global object=] of [=this=] using the [=gamepad task source=] to
+            run the following steps:
+              <ol>
+                <li>[=Resolve=]
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+                with {{GamepadHapticsResult/"preempted"}}.
+                </li>
+                <li>Set
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
+                `undefined`.
+                </li>
+              </ol>
+            </li>
+            <li>[=Stop haptic effects=] on this actuator.
+            </li>
+            <li>Return [=a promise resolved with=]
```

Comment:
The promise should probably only resolve once it gets confirmation over IPC that the actuator has actually stopped.